src/foam/nanos/auth/AgentUserAuthService.java

```diff
@@ -100,6 +87,46 @@ public User actAs(X x, User entity) throws AuthenticationException {
     return user;
   }
 
+  public boolean canActAs(X x, User agent, User entity) {
+    try {
+      Group group = (Group) groupDAO_.find(entity.getGroup());
+
+      if ( group == null ) {
+        throw new AuthorizationException("Entity must exist within a group.");
+      }
+
+      if ( ! group.getEnabled() ) {
+        throw new AuthorizationException("Entity' group must be enabled.");
+      }
+
+      /*
+      Finds the UserUserJunction object to see if user can act as the passed in user.
+      Source (agent) users are permitted to act as target (entity) users, not vice versa.
+      */
+      UserUserJunction permissionJunction = (UserUserJunction) agentJunctionDAO_.find(AND(
```

Comment:
Isn't there a method on the relationship or in the object to access the junction more conveniently?

Reply:
I think you could do something like:
 agent.getEntities().getJunctionDAO().find(entity.getId())
or maybe just:
 agent.getEntities().getJunctionDAO().find(entity)

Which is shorter and doesn't require you know the junction key or the method of joining with agent.

Reply:
We're trying to the junction object, and not the entity - it has a status and a group associated with it that we need to check and assign respectively.

Reply:
Is there a way to do that through relationships?

Reply:
agent.getEntities().getJunctionDAO() gets the JunctionDAO, which contains the junctions.

Reply:
sure, but then the rest of the query will be the same right? Happy to change if you'd like, but doesn't seem super consequential to me. If anything, the way it is might be marginally faster? Let me know either way.